docs/spec/sources.md

```diff
@@ -0,0 +1,342 @@
+                    type:
+                      type: string
+                      pattern: "dev.knative.apiserver.resource.add"
+                    schema:
+                      type: string
+                deleteResource:
+                  type: object
+                  properties:
+                    type:
+                      type: string
+                      pattern: "dev.knative.apiserver.resource.delete"
+                    schema:
+                      type: string
+                updateResource:
+                  type: object
+                  properties:
+                    type:
+                      type: string
+                      pattern: "dev.knative.apiserver.resource.update"
+                    schema:
+                      type: string
+                ...
+```
+-->
+
+## Source RBAC
+
+Sources can by any shape and are expected to be extensions onto Kubernetes. To
+prevent cluster operators from re-deploying, or re-creating RBAC for all
+accounts that will interact with sources, Knative leverages aggregated RBAC
+roles to dynamically update the rights of controllers that are using common
+service accounts provided by Knative. This allows Eventing controllers to check
+the status of source type resources without being aware of the exact source type
+or implementation at compile time.
+
+The
+[`source-observer` ClusterRole](../../config/200-source-observer-clusterrole.yaml)
+looks like:
+
+```yaml
+# Use this aggregated ClusterRole when you need read "Sources".
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: source-observer
+aggregationRule:
+  clusterRoleSelectors:
+    - matchLabels:
+        duck.knative.dev/source: "true"
+rules: [] # Rules are automatically filled in by the controller manager.
+```
+
+And new sources MUST include a ClusterRole as part of installing themselves into
+a cluster:
+
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: foos-source-observer
+  labels:
+    duck.knative.dev/source: "true"
+rules:
+  - apiGroups:
+      - example.com
+    resources:
+      - foos
+    verbs:
+      - get
+      - list
+      - watch
+```
+
+## Source Custom Objects
+
+The minimum definition of the Kubernetes resource shape is defined in the
+[Source](https://godoc.org/github.com/knative/pkg/apis/duck/v1#Source) ducktype.
+
+### duck.Spec
+
+The `spec` field is expected to have the following minimum shape:
+
+```go
+type SourceSpec struct {
+    // Sink is a reference to an object that will resolve to a domain name or a
+    // URI directly to use as the sink.
+    Sink apisv1alpha1.Destination `json:"sink,omitempty"`
+
+    // CloudEventOverrides defines overrides to control the output format and
+    // modifications of the event sent to the sink.
+    // +optional
+    CloudEventOverrides *CloudEventOverrides `json:"ceOverrides,omitempty"`
+}
+```
+
+For a full definition of `Sink` and `CloudEventsOverrides`, please see
```

Comment:
I think you mean "a golang structure definition" rather than "a full definition". I'd expect a full definition to be a documentation file explaining both the structure and rationale, rather than header docs.